src/openshift/component.ts

```diff
@@ -67,13 +67,24 @@ export class Component extends OpenShiftItem {
         const name: string = component.getName();
         const value = await window.showWarningMessage(`Do you want to delete Component '${name}\'?`, 'Yes', 'Cancel');
         if (value === 'Yes') {
+            await Component.unlinkComponent(component);
             return Progress.execFunctionWithProgress(`Deleting the Component '${component.getName()} '`, () => {
                 return Component.odo.deleteComponent(component);
             }).then(() => `Component '${name}' successfully deleted`)
                 .catch((err) => Promise.reject(`Failed to delete Component with error '${err}'`));
         }
     }
 
+    static async unlinkComponent(component: OpenShiftObject) {
+        const compData = await Component.odo.execute(Command.getdescribeComponent(component.getParent().getParent().getName(), component.getParent().getName(), component.getName()));
+        const compObj: JSON = JSON.parse(compData.stdout);
+        if (compObj['status'].linkedComponents) {
```

Comment:
@sudhirverma try to use map function here and pass the ket to `unlinkComponentTo`.